app/goal_routes.py

```diff
@@ -0,0 +1,106 @@
+from flask import Blueprint, json, jsonify, request
+from app.models.goal import Goal
+from app.models.task import Task
+from app import db
+from datetime import datetime
+
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+@goals_bp.route("", methods=["POST"])
+def create_goal():
+    request_body = request.get_json()
+
+    if 'title' not in request_body:
+        return {"details": "Invalid data"}, 400
+
+    new_goal = Goal(
+        title=request_body['title']
+    )
+
+    db.session.add(new_goal)
+    db.session.commit()
+
+    return {"goal": new_goal.to_dict()}, 201
+
+@goals_bp.route("/<goal_id>/tasks", methods=["POST"])
+def create_tasks_for_goal(goal_id):
```

Comment:
Here and throughout your code you've got excellent, very descriptive function names!